flume-ng-sinks/flume-http-sink/src/main/java/org/apache/flume/sink/http/HttpSink.java

```diff
@@ -0,0 +1,408 @@
+
+      // re-throw all Errors
+      if (t instanceof Error) {
+        throw (Error) t;
+      }
+
+    } finally {
+      txn.close();
+
+      if (outputStream != null) {
+        try {
+          outputStream.close();
+        } catch (IOException e) {
+          // ignore errors
+        }
+      }
+    }
+
+    return status;
+  }
+
+  /**
+   * Reads a set of override values from the context configuration and stores
+   * the results in the Map provided.
+   *
+   * @param propertyName  the prefix of the config property names
+   * @param context       the context to use to read config properties
+   * @param override      the override Map to store results in
+   */
+  private void parseConfigOverrides(final String propertyName,
+                                    final Context context,
+                                    final Map<String, Boolean> override) {
+
+    ImmutableMap<String, String> config = context.getSubProperties(
+        propertyName + ".");
+
+    if (config != null) {
+      for (Map.Entry<String, String> value : config.entrySet()) {
+        LOG.info(String.format("Read %s value for status code %s as %s",
+            propertyName, value.getKey(), value.getValue()));
+
+        if (override.containsKey(value.getKey())) {
+          LOG.warn(String.format("Ignoring duplicate config value for %s.%s",
+              propertyName, value.getKey()));
+        } else {
+          override.put(value.getKey(), Boolean.valueOf(value.getValue()));
+        }
+      }
+    }
+  }
+
+  /**
+   * Queries the specified override map to find the most appropriate value. The
+   * most specific match is found.
+   *
+   * @param statusCode    the String representation of the HTTP status code
+   * @param overrides     the map of status code overrides
+   * @param defaultValue  the default value to use if no override is configured
+   *
+   * @return the value of the most specific match to the given status code
+   */
+  private boolean findOverrideValue(final String statusCode,
+                                    final HashMap<String, Boolean> overrides,
+                                    final boolean defaultValue) {
+
+    Boolean overrideValue = overrides.get(statusCode);
+    if (overrideValue == null) {
+      overrideValue = overrides.get(statusCode.substring(0, 1) + "XX");
+      if (overrideValue == null) {
+        overrideValue = defaultValue;
+      }
+    }
+    return overrideValue;
+  }
+
+  /**
+   * Update the connection builder.
+   *
+   * @param builder  the new value
+   */
+  final void setConnectionBuilder(final ConnectionBuilder builder) {
+    this.connectionBuilder = builder;
+  }
+
+  /**
+   * Update the sinkCounter.
+   *
+   * @param newSinkCounter  the new value
+   */
+  final void setSinkCounter(final SinkCounter newSinkCounter) {
+    this.sinkCounter = newSinkCounter;
+  }
+
+  /**
+   * Class used to allow extending the connection building functionality.
+   */
+  class ConnectionBuilder {
```

Comment:
as mentioned earlier, could we try to re-user HttpURLConnection, if the end point has not been changed in configuration file and status is still good?